src/main/java/cat/udl/eps/softarch/linkapp/repository/MessageRepository.java

```diff
@@ -0,0 +1,15 @@
+package cat.udl.eps.softarch.linkapp.repository;
+
+import cat.udl.eps.softarch.linkapp.domain.Message;
+import cat.udl.eps.softarch.linkapp.domain.User;
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.data.repository.query.Param;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+import java.util.List;
+
+@RepositoryRestResource
+public interface MessageRepository extends PagingAndSortingRepository<Message, Long> {
+  public List<Message> findByAuthor(@Param("author") User author);
+  public Message findByIdContaining(@Param("number") Long number);
```

Comment:
These methods are not tested.

Reply:
findByAuthor has been tested when checking that the author of a new message is the correct one. Regarding the other method, we've removed it as it was useless considering that it was never used and we have the findById predefined method.